cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java

```diff
@@ -626,4 +628,86 @@ protected void kafkaHeapOptions(List<EnvVar> envVars, double dynamicHeapFraction
         }
     }
+
+    public static int getInteger(Map<String, String> data, String key, int defaultValue) {
+        try {
+            if (data.get(key) == null) { // value is not set in config map -> will be used default value
+                return defaultValue;
+            }
+            return Integer.parseInt(data.get(key));
+        } catch (NumberFormatException e)  {
+            String msg = " is corrupted";
+            if (data.get(key).isEmpty()) {
+                msg = " is empty";
+            }
+            throw new InvalidConfigMapException(key, msg);
+        }
+    }
+
+    public static Storage getStorage(Map<String, String> data, String key) {
+        try {
+            if (data.get(key) == null) { // value is not set in config map
+                return new Storage(Storage.StorageType.PERSISTENT_CLAIM);
+            }
+            return Storage.fromJson(new JsonObject(data.get(key)));
+        } catch (Exception e) {
+            throw new InvalidConfigMapException(key, " is corrupted");
+        }
+    }
+
+    public static String getNonemptyString(Map<String, String> data, String key, String defaultValue) {
+        if (data.get(key) == null) { // value is not set in config map -> will be used default value
+            return defaultValue;
+        }
+        if (data.get(key).length() == 0) {
+            throw new InvalidConfigMapException(key, " is empty");
+        }
+        return data.get(key);
+    }
+
+    public static String getJsonCorruptionBlame(DecodeException de, String config) {
+        String token = "";
+        token = de.getMessage();
+        if (token.contains("Illegal unquoted character")) {
+            return "JSON quotation";
+        }
+        token = token.replace("Failed to decode: Unrecognized token '", "");
+        token = token.substring(0, token.indexOf("'"));
+
+        String[] entries = config.replace("{", "").replace("}", "")
+                .replace("\"", "").replace(" ", "")
+                .replace("\n", "").split(",");
+        int i;
+        String blame = "";
+        for (i = 0; i < entries.length; ++i) {
+            if (entries[i].split(":")[1].equals(token)) {
+                blame = entries[i].split(":")[0];
+                break;
+            }
+        }
+        return blame;
+    }
+
+    public static JsonObject getConfig(Map<String, String> data, String key) {
+        String config = "";
+        try {
+            config = data.get(key);
+            Map<String, Object> map = new JsonObject(config).getMap();
+            Map<String, String> newMap = new HashMap<String, String>();
+            for (Map.Entry<String, Object> entry : map.entrySet()) {
+                if (entry.getValue() instanceof String) {
+                    newMap.put(entry.getKey(), (String) entry.getValue());
+                } else {
+                    throw new InvalidConfigMapException(entry.getKey(), " has JSON with key " + entry.getKey() + " with a non-String value");
```

Comment:
The JSON string with the config expects to receive also booleans, floats etc. These should be properly converted.